src/transformState.js

```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
```

Comment:
this can also be achieved using [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) which is more common nowadays

Reply:
Are you sure? The state object should be mutable, and the spread operator creates a copy of it without mutating it. I also cannot change the values of the state because parameters of a function cannot be reassigned, in my opinion its the best way to do

Reply:
That's correct, it's very nice that you provided an argument.